src/Modules/Authontication/Components/LogIn/LogIn.tsx

```diff
@@ -1,7 +1,144 @@
-import React from 'react'
+import { useForm } from 'react-hook-form';
+
+import { useContext, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { USERS_URLs } from '../../../../constans/END_POINTS';
+import { AuthContext } from '../Context/AuthContext'
+import { EmailValidation, PasswordValidation } from '../../../../constans/VALIDATIONS';
 
 export default function LogIn() {
+  let { saveLoginData } = useContext(AuthContext);
+  let navigate = useNavigate();
+  let{
+    register,
+    handleSubmit,
+    formState:{errors},
+  } = useForm({ mode: "onBlur" });
+
+
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const toggleVisibility = (setterFunction: any) => {
+    setterFunction((prevState: any) => !prevState);
+  };
+
+  let onSubmit = async (data:any)=>{
+    try {
+      let response = await axios.post(USERS_URLs.Login, data);
```

Comment:
We need to properly type the response as mentioned before.

Reply:
cannot understand what do you mean??

Reply:
```ts
post<ResponseType>(USERS_URLs.Login, data)
```
We can get the response type structure from the backend response by including the fields that we are interested in.

This will type the response for us and enable us to access the proper values and get auto completion for them when pressing the space bar.